File: 1-easy/problems/0965-univalued-binary-tree/solution.ts
function isUnivalTree(root: any): boolean | number | string | any {
  """
  :type root: TreeNode
  :rtype: bool
  """
  s = [root]
  while (s) {
      node = s.pop()
      if (not node) {
          continue
      if (node.val != root.val) {
          return false;
      s.append(node.left)
      s.append(node.right)
  return true;
}

export { isUnivalTree };

File: 1-easy/problems/0965-univalued-binary-tree/solution.py
class Solution:
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        s = [root]
        while s:
            node = s.pop()
            if not node:
                continue
            if node.val != root.val:
                return False
            s.append(node.left)
            s.append(node.right)
        return True
    

class Solution2(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        return (not root.left or (root.left.val == root.val and self.isUnivalTree(root.left))) and \
               (not root.right or (root.right.val == root.val and self.isUnivalTree(root.right)))
